celltype_annotation/analysis/singler-rms-comparison.Rmd

```diff
@@ -0,0 +1,245 @@
+    dplyr::mutate(reference_factor = forcats::fct_relevel(reference, reference_order))
+  
+  return(celltype_assignments)
+}
+```
+
+## Overall celltype assignments 
+
+This section examines the overall distribution of cell type assignments across all references with `SingleR` and manual annotations.
+We look at this using a stacked bar chart and by printing out a table of cell type annotations.
+
+The original submitter cell types contained sub groups of tumor cells, e.g., `Tumor_myoblast-A` and `Tumor_myoblast-B`. 
+Before comparing annotations, the subgroups are combined into a single cell type, e.g., `Tumor_myoblast`.
+These will be stored in the `broad_celltype` column and used throughout the remainder of the notebook. 
+
+```{r}
+# extract coldata for easy plotting
+coldata_df <- colData(singler_annotated_sce) |>
+  as.data.frame() |> 
+  # classify cells as tumor or normal
+  dplyr::mutate(cell_category = dplyr::if_else(stringr::str_detect(submitter_annotations, "Tumor"), "Tumor", submitter_annotations),
+                # remove sub states of myoblast/mesoderm/myocyte
+                broad_celltype = stringr::str_remove(submitter_annotations, "-A|-B|-C|-D"))
+```
+
+For all the `label.ont` reference labels, we will grab the human readable cell ontology labels.
+These labels will be used for plotting for the remainder of the notebook. 
+
+```{r}
+# grab cell label names corresponding to cell ontology ids
+cl_ont <- ontoProc::getCellOnto()
+
+# rename all label.ont columns with full cell ontology label
+ont_columns_idx <- startsWith(colnames(coldata_df), "label.ont")
+cl_columns <- coldata_df[,ont_columns_idx] |>
+  purrr::map(
+    \(cl_id) 
+    cl_ont$name[cl_id]
+  ) |> 
+  dplyr::bind_cols()
+coldata_df[,ont_columns_idx] <- cl_columns
+```
+
+```{r}
+# create a table of cell type assignments for each label type
+labels <- c("label.main", "label.fine", "label.ont")
+celltype_assignment_list <- purrr::map(labels,
+                                       \(label_type) celltype_assignments_table(coldata_df,
+                                                                                label_type)) |>
+  purrr::set_names(labels)
+```
+
+
+```{r, fig.width=10, fig.height=5}
+# print out barcharts
+purrr::imap(celltype_assignment_list, 
+           \(celltype_assignments, label_type) celltype_barchart(celltype_assignments,
+                                                                 label_type))
+```
+
+```{r}
+# print out tables 
+purrr::map(celltype_assignment_list, 
+            \(celltype_assignments) 
+            celltype_table(celltype_assignments))
+```
+
+## Comparison of SingleR annotations to submitter annotations
+
+This section will directly compare cell type annotations using `HumanPrimaryCellAtlasData` or `BlueprintEncodeData` with `SingleR` vs. manual annotations. 
+Each chunk displays a heatmap directly comparing a `SingleR` reference (columns) to manual annotations (rows).
```

Comment:
Is there a way to add titles to these plots when they are printed for which reference is shown? Will be really helpful for easier viewing/interpretation!